breakdown/context7.md

```diff
@@ -0,0 +1,949 @@
+
+    Rank --> QuestEval
+    Rank --> LLMEval
+    Rank --> FormatVal
+    Rank --> MetaFilter
+    Rank --> InitCheck
+
+    classDef important fill:#ff6b6b,stroke:#d63031,stroke-width:3px
+    classDef processing fill:#e8f5e9,stroke:#2e7d32,stroke-width:2px
+    classDef quality fill:#e1f5fe,stroke:#01579b,stroke-width:2px
+    classDef sources fill:#fff3e0,stroke:#ef6c00,stroke-width:2px
+
+    class MCP,C7API important
+    class Parse,Enrich,Vector,Rank,Cache processing
+    class QuestEval,LLMEval,FormatVal,MetaFilter,InitCheck quality
+    class GitHub,NPM,PyPI,Maven,Other_Reg sources
+```
+
+### Request flow
+
+Under the hood, Context7 orchestrates a carefully designed sequence that transforms outdated LLM knowledge into current, working code:
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant Client as MCP Client
+    participant Server as Context7 Server
+    participant Handler as Tool Handler
+    participant API as Context7 API
+    participant LLM
+
+    User->>Client: "Create Next.js app. use context7"
+    Client->>Server: MCP connection (stdio/http/sse)
+    Client->>Server: Detect "use context7" trigger
+
+    Note over Server: Tool Resolution Phase
+    Server->>Handler: CallToolRequest("resolve-library-id")
+    Handler->>API: searchLibraries("next.js")
+    API-->>Handler: [{id: "/vercel/next.js", trust: 8.5}]
+    Handler-->>Server: CallToolResult with library ID
+
+    Note over Server: Documentation Fetch Phase
+    Server->>Handler: CallToolRequest("get-library-docs")
+    Handler->>API: fetchLibraryDocumentation("/vercel/next.js", {topic: "app router"})
+    API-->>Handler: Current Next.js 15 docs (filtered, ranked)
+    Handler-->>Server: CallToolResult with documentation
+
+    Server-->>Client: Enhanced context with docs
+    Client->>LLM: Original prompt + injected documentation
+    LLM-->>Client: Response with current, working code
+    Client-->>User: Accurate Next.js 15 implementation
+```
+
+### Transport mechanisms
+
+Context7's transport layer is fundamentally adaptive - it detects your client and chooses the optimal communication method:
+
+```typescript
+// Actual CLI argument processing from Context7 MCP
+const program = new Command()
+  .option("--transport <stdio|http|sse>", "transport type", "stdio")
+  .option("--port <number>", "port for HTTP/SSE transport", "3000")
+  .allowUnknownOption() // let MCP Inspector / other wrappers pass through
+  .parse(process.argv);
+
+// Validate and set transport type
+const allowedTransports = ["stdio", "http", "sse"];
+if (!allowedTransports.includes(cliOptions.transport)) {
+  console.error(
+    `Invalid --transport value: '${cliOptions.transport}'. Must be one of: stdio, http, sse.`
+  );
+  process.exit(1);
+}
+
+// The clever bit: Auto-detection based on client
+const TRANSPORT_TYPE = (cliOptions.transport || "stdio") as
+  | "stdio" // Cursor, Claude Desktop
+  | "http" // VS Code extensions
+  | "sse"; // Windsurf, streaming clients
+```
```

Comment:
Is this related to Context7 or was there an idea to connect the dots with MCP a bit more here?